### ISSUE TITLE
cannot load such file -- ci/reporter/rake/rspec

### ISSUE THREAD

**Issue by ryanjaeb** [jbox-web/redmine_jenkins#16]
Hi,

While trying to run `rake redmine:plugins:migrate RAILS_ENV=production` after installing the plugin I got the following error:

```
rake aborted!
cannot load such file -- ci/reporter/rake/rspec
```

I create my Redmine bundle using the following two commands:

```
bundle install --path vendor/bundle --without development test mysql sqlite rmagick --no-deployment
bundle install --deployment
```

I assume that since I'm not using the `development` or `test` profile, I shouldn't need any of the `rspec` or `ci` related stuff.  Is it possible to change the scope of the `ci/reporter/rake/rspec` requirement in `lib/tasks/redmine_jenkins.rake`?  I tried the following and didn't get any errors, but I'm not familiar enough with ruby / rails to know if it's the correct solution.

```
namespace :redmine_jenkins do

  namespace :ci do
    require 'ci/reporter/rake/rspec'

    ENV["CI_REPORTS"] = Rails.root.join('junit').to_s
    RSpec::Core::RakeTask.new do |task|
      task.rspec_opts = "plugins/redmine_jenkins/spec --color"
    end
    task :all => ['ci:setup:rspec', 'spec']
  end


  desc "Show library version"
  task :version do
    puts "Redmine Jenkins #{version("plugins/redmine_jenkins/init.rb")}"
  end


  def version(path)
    line = File.read(Rails.root.join(path))[/^\s*version\s*.*/]
    line.match(/.*version\s*['"](.*)['"]/)[1]
  end

end
```

**Comment by n-rodriguez** [comment]
Thanks for your feedback!

**Comment by ryanjaeb** [comment]
No problem.  Thanks for the plugin :-)

**Comment by ryanjaeb** [comment]
I just noticed this didn't actually solve my problem.  I didn't realize until just now, but I had moved the `rspec` related gems in `remdine_jenkins/Gemfile` outside of the `:development, :test` scope and forgot to move them back.

I don't think the location of that require line actually matters, but, as I mentioned, I don't know much about ruby.  I can give a half decent explanation of the problem I'm having though.  The `Gemfile` for the plugin is declaring the `rspec` related dependencies for the `test` and `development` ENV, but the `lib/tasks/redmine_jenkins.rake` is doing something in the `:ci` section that needs the `rspec` stuff.

Since I'm only bundling for the `production` ENV those gems don't exist and the `rake` command fails.  Deleting the `require` line plus the whole `namespace :ci` section from the `redmine_jenkins.rake` file is enough that I can actually get the plugin installed.

Let me know if I can provide any more info.
